src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -1816,9 +1816,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
             Dim sourceType As ITypeSymbol = source.Type
 
+            Dim sourceConstantValue as ConstantValue = source.GetConstantValue()
             If sourceType Is Nothing Then
-                If source.ConstantValue.HasValue AndAlso source.ConstantValue.Value Is Nothing AndAlso destination.IsReferenceType Then
-                    constantValue = source.ConstantValue
+                If sourceConstantValue?.IsNothing = True AndAlso destination.IsReferenceType Then
```

Comment:
>sourceConstantValue?.IsNothing = True [](start = 19, length = 37)

I don't think this does short-circuiting on Null in VB, consider doing explicit ```IsNot Nothing``` check, followed by IsNothing check